tests/lib/systemd.sh

```diff
@@ -18,3 +18,11 @@ systemd_stop_and_destroy_unit() {
     rm -f "/run/systemd/system/$1.service"
     systemctl daemon-reload
 }
+
+wait_for_service() {
+    local service_name="$1"
+    local state="${2:-active}"
+    while ! systemctl show -p ActiveState "$service_name" | grep -q "ActiveState=$state"; do
```

Comment:
This will spam the log and run forever if the service is not up in some reasonable amount of time. Assuming we have 10 minute timeout on individual test this would print, 600 times, about a dozen lines.

I was wondering if we could instead require a timeout and only wait so-many-times, printing the status only if the unit eventually fails.

Reply:
I added code to limit the debug output and also to have a build in timeout for the service start now.